main.go

```diff
@@ -107,16 +107,23 @@ func runExtension(ctx context.Context, logger *log.Logger, wg *sync.WaitGroup) (
 		return nil, fmt.Errorf("nil client returned: %w", err)
 	}
 
+	var newState string
+	//Leverage Vault helpers for eventual consistency on login
+	client.VaultClient = client.VaultClient.WithResponseCallbacks(api.RecordState(&newState)).WithRequestCallbacks(api.RequireState(newState))
```

Comment:
`newState` is passed to the request callback by value before getting set, so I don't think this will quite work yet. It also asks Vault to do a little more work than it needs to if the extension is configured to e.g. read dynamic secrets, because each dynamic secret will return a new state and the next secret read will wait for that state to be replicated.

Could we only set a response callback on this line, and then clear that response callback and add a request callback on L118 below? Alongside the clearing of all callbacks at L125, I think that will have all the bases covered then.